lib/connections/tcp_dial.go

```diff
@@ -33,7 +34,9 @@ type tcpDialer struct {
 func (d *tcpDialer) Dial(id protocol.DeviceID, uri *url.URL) (internalConn, error) {
 	uri = fixupPort(uri, config.DefaultTCPPort)
 
-	conn, err := dialer.DialTimeout(uri.Scheme, uri.Host, 10*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	conn, err := dialer.DialContextReusePort(ctx, uri.Scheme, uri.Host)
+	cancel()
```

Comment:
`dialer` is used below, so defer?

Reply:
dialer is a package name in this case, so cancel just releases resources associated with the context constructed above post dial